### PR TITLE
Add script to generate voting public key and voting authorization

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
Current dependencies on/for this PR:
* `feature/proof-of-stake`
  * **PR #988** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/988?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/988?utm_source=stack-comment).

File: scripts/pos/validator_registration_utils.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	return pubKey, votingAuthorization
+}
+
+// You must have relic installed to run this code.
+// To install relic, use the install-relic.sh script in the scripts directory.
+// go run -tags relic validator_registration_utils.go
+func main() {
+	// Replace with your own BLS private key hex.
+	// blsPrivateKey, err := bls.PrivateKey{}.FromString("...")
+	blsPrivateKey, err := bls.NewPrivateKey()
```
Comment (tholonious) [inline scripts/pos/validator_registration_utils.go:33]:
```suggestion
	// Replace with your own BIP39 Validator seed phrase
	keystore, err := lib.NewBLSKeystore("...")
```

File: scripts/pos/validator_registration_utils.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+func getBLSVotingAuthorizationAndPublicKey(blsPrivateKeyHex string) (*bls.PublicKey, *bls.Signature) {
+	// TODO: seed phrase or private key hex?
+	// privKey, err := bls.PrivateKey{}.FromSeed(blsPrivateKeyHex)
+	privKey, err := (&bls.PrivateKey{}).FromString(blsPrivateKeyHex)
+	if err != nil {
+		panic(err)
+	}
+	pubKey := privKey.PublicKey()
```
Comment (tholonious) [inline scripts/pos/validator_registration_utils.go:18]:
```suggestion
func getBLSVotingAuthorizationAndPublicKey(keystore *lib.BLSKeystore) (*bls.PublicKey, *bls.Signature) {
	pubKey := keystore.GetPublicKey()
```

File: scripts/pos/validator_registration_utils.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	votingAuthPayload := lib.CreateValidatorVotingAuthorizationPayload(pubKey.ToBytes())
+	votingAuthorization, err := privKey.Sign(votingAuthPayload)
```
Comment (tholonious) [inline scripts/pos/validator_registration_utils.go:20]:
```suggestion
	votingAuthorization, err := keystore.GetSigner().Sign(votingAuthPayload)
```

File: scripts/pos/validator_registration_utils.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+// go run -tags relic validator_registration_utils.go
+func main() {
+	// Replace with your own BLS private key hex.
+	// blsPrivateKey, err := bls.PrivateKey{}.FromString("...")
+	blsPrivateKey, err := bls.NewPrivateKey()
+	if err != nil {
+		panic(err)
+	}
+	blsPrivateKeyHex := blsPrivateKey.ToString()
+	publicKey, votingAuthorization := getBLSVotingAuthorizationAndPublicKey(blsPrivateKeyHex)
```
Comment (tholonious) [inline scripts/pos/validator_registration_utils.go:38]:
```suggestion
	publicKey, votingAuthorization := getBLSVotingAuthorizationAndPublicKey(keystore)
```

**Review by tholonious — CHANGES_REQUESTED** [review]
You can generate the lib.BLSKeystore from a BIP39 seed phrase and use it to sign the voting authorization!

**Pull request closed without merging**